packages/react-components/src/components/chart/hooks/useEvents.ts

```diff
@@ -0,0 +1,135 @@
+import { useEffect } from 'react';
+import { EChartsType, getInstanceByDom } from 'echarts';
+import { MAX_TREND_CURSORS, TREND_CURSOR_CLOSE_GRAPHIC_INDEX } from '../eChartsConstants';
+import { calculateTimeStamp } from '../utils/getInfo';
+import { v4 as uuid } from 'uuid';
+import { getNewTrendCursor, ondragUpdateTrendCursor } from '../utils/getTrendCursor';
+import { UseEventsProps } from '../types';
+import useDataStore from '../../../store';
+
+let trendCursorStaticIndex = 0;
+
+const useEvents = ({
```

Comment:
Why is it called `useEvents`? Is there more than trend cursors?

Reply:
TCs needs to handle user events, resize and sync... and hence 3 different hooks